delphix/virtualization-sdk
# CP-9575 Add support for additional mount in vsdk/dvp for dSources operations

### Background
Currently, Delphix engine does not support multiple mounts for dSource created by AppData / plugins / select connectors. For cluster based databases, the requirement is to support additional mount points for dSources also.
As part of the changes, `mounts` is added to `staged_source` and `StagedMountSpecResponse`.

### Problem
IDEA is to support multiple mounts for Staged sources.

### Solution
dvp-api 1.7.1.dev0 is published to TestPyPi which now support `mounts` in `staged_source` and `StagedMountSpecResponse`.

`Platform` package has been updated to now consume the `mounts` from `staged_source` and also set the multiple mounts received from plugin to `StagedMountSpecResponse`.

### Testing Done
*Manual Testing*
- added unit test cases to cover all the changes done as part of the solution. Now the test cases for staged source covers single mount, multiple mount and both parameters set in staged_source.
- Manually Tested the code for older plugins (works fine with single mount point) and also tested with multiple mount points plugins (for newer changes).

*Automated Testing*
For auomated testing, we are only testing for regression use cases as we are supporting backward compatibility, so older plugins should work fine with the new changes.

- appdata_python_samples - http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74193

- Appdata Basic 
    - [nix_staged_python - RHEL 7.9](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74212)
    - [Appdata Python win_staged_python - Win 2019](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74213)
    - [Appdata Python nix_direct_python - CentOS 8.0](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74211)

- Virtualisation SDK
    - [Nix direct appdata python plugin](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74194)
    - [Nix staged appdata python plugin](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74195)
    - [Windows direct appdata python plugin](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74196)
    - [Windows staged appdata python plugin](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74197)
    - [Windows staged default plugin built on Windows 10 Pro SDK host](http://selfservice.jenkins.delphix.com/job/blackbox-self-service/74198)
